universal-login-relayer/lib/services/WalletService.js

```diff
@@ -1,11 +1,50 @@
 import WalletContract from 'universal-login-contracts/build/WalletContract';
 import LegacyWallet from 'universal-login-contracts/build/LegacyWallet';
-import {hasEnoughToken, isAddKeyCall, getKeyFromData, isAddKeysCall} from '../utils/utils';
-import {utils, ContractFactory} from 'ethers';
+import {hasEnoughToken, isAddKeyCall, getKeyFromData, isAddKeysCall, sortExecutionsByKey} from '../utils/utils';
+import {utils, ContractFactory, Contract} from 'ethers';
 import defaultDeployOptions from '../config/defaultDeployOptions';
+import {concatenateSignatures, calculateMessageHash} from 'universal-login-contracts';
 
+class PendingExecution {
+
+  constructor(walletAddress, wallet) {
+    this.Wallet = wallet;
+    this.WalletContract = new Contract(walletAddress, WalletContract.interface, this.Wallet);
+    this.CollectedSignatures = [];
+  }
+
+  async getStatus() {
+    return {
+      collectedSignatures: this.CollectedSignatures,
+      totalCollected: this.CollectedSignatures.length,
+      required: await this.WalletContract.requiredSignatures()
+    };
+  }
+
+  async push(from, to, value, data, nonce, gasPrice, gasToken, gasLimit, operationType, signature) {
```

Comment:
We should pack all those fields into an object. This object already exist and is called `Message`. It is currently defined here:
https://github.com/UniversalLogin/UniversalLoginSDK/blob/master/universal-login-sdk/lib/sdk.ts

Should be moved to commons.

Than wee can do:
```js
async push(message) { 
  ...
}
```